src/app/child-dev-project/attendance/attendance.service.ts

```diff
@@ -5,17 +5,106 @@ import { Note } from "../notes/model/note";
 import { RecurringActivity } from "./model/recurring-activity";
 import { ActivityAttendance } from "./model/activity-attendance";
 import { groupBy } from "../../utils/utils";
+import { DatabaseIndexingService } from "../../core/entity/database-indexing/database-indexing.service";
+import { ConfigService } from "../../core/config/config.service";
+import {
+  INTERACTION_TYPE_CONFIG_ID,
+  InteractionType,
+} from "../notes/model/interaction-type.interface";
 
 @Injectable({
   providedIn: "root",
 })
 export class AttendanceService {
-  constructor(private entityMapper: EntityMapperService) {}
+  constructor(
+    private entityMapper: EntityMapperService,
+    private dbIndexing: DatabaseIndexingService,
+    private configService: ConfigService
+  ) {
+    this.createIndices();
+  }
+
+  private async createIndices() {
+    const meetingInteractionTypes = this.configService
+      .getConfig<InteractionType[]>(INTERACTION_TYPE_CONFIG_ID)
+      .filter((t) => t.isMeeting)
+      .map((t) => t.id);
+    await this.createEventsIndex(meetingInteractionTypes);
+    await this.createRecurringActivitiesIndex();
```

Comment:
Is there a reason why `await` is used here? This prevents the indices from being built in parallel and the using function doesn't wait on the result anyway.

Reply:
I am using this method in tests to actually await completion - otherwise there are crashes due to the database being torn down too early.

Running index creations in parallel or sequentially has almost the same performance, see these stats:
https://github.com/Aam-Digital/ndb-core/issues/462#issuecomment-651891399